server/sonar-process/src/main/java/org/sonar/process/cluster/info/ProcessInfoProviderRef.java

```diff
@@ -17,24 +17,25 @@
  * along with this program; if not, write to the Free Software Foundation,
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
-package org.sonar.server.platform.monitoring;
+package org.sonar.process.cluster.info;
 
-import java.util.Map;
 import java.util.Objects;
-import java.util.TreeMap;
+import javax.annotation.Nullable;
 
-public class JvmPropsMonitor implements Monitor {
-  @Override
-  public String name() {
-    return "JvmProperties";
+/**
+ * Trick to be able to access implementation of {@link ProcessInfoProvider}
+ * from the Hazelcast members that execute {@link ProcessInfoCall}.
+ */
+public class ProcessInfoProviderRef {
+
+  private static ProcessInfoProvider processInfoProvider = null;
+
+  public static ProcessInfoProvider getProcessInfoProvider() {
+    Objects.requireNonNull(processInfoProvider, "ProcessInfoProvider is not available");
+    return processInfoProvider;
   }
 
-  @Override
-  public Map<String, Object> attributes() {
-    Map<String, Object> sortedProps = new TreeMap<>();
-    for (Map.Entry<Object, Object> systemProp : System.getProperties().entrySet()) {
-      sortedProps.put(Objects.toString(systemProp.getKey()), Objects.toString(systemProp.getValue()));
-    }
-    return sortedProps;
+  public static void setProcessInfoProvider(@Nullable ProcessInfoProvider p) {
+    processInfoProvider = p;
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Synchronize this lazy initialization of 'processInfoProvider' [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2444)